### ISSUE TITLE
Can not load i18n files if using requireJS

### ISSUE THREAD

**Issue by JanMalte** [select2/select2#2899]
Having non english language settings makes select2 fail to load the language files if using requireJS.

For a german user select2 (4.0.0-beta) is not working and the error printed in the browser console is the following:
```
No select2/i18n/de
```

I do use bower and requireJS to manage all libraries for my project. Therefor all libraries are located under ```static/assets/vendor/...```

**bower.json**
```json
{
  "name": "myproject",
  "version": "0.0.1",
  "dependencies": {
    "modernizr": "~2.8.3",
    "bootstrap-sass-official": "~3.2",
    "jquery": "~2.1",
    "requirejs": "~2.1.15",
    "requirejs-domready": "~2.0.1",
    "select2": "~4.0.0",
  },
}
```

**.bowerrc**
```
{
    "directory": "vendor"
}
```

**main.js**
```js
require.config({
    paths: {
        'domReady': ['//cdnjs.cloudflare.com/ajax/libs/require-domReady/2.0.1/domReady', '../vendor/requirejs-domready/domReady'],
        'jQuery': ['//ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min', '../vendor/jquery/dist/jquery.min'],
        'bootstrap': '../vendor/bootstrap-sass-official/assets/javascripts/bootstrap',
        'select2': '../vendor/select2/dist/js/select2.min',
    },
    "shim": {
        'jQuery': {
            exports: 'jQuery'
        },
    }
})

// Load and enable bootstrap
require(['domReady!', 'jQuery', 'select2'], function() {
    $('.select2').not('.select2-container').select2();
});
```

**Comment by MrTool** [comment]
I got the same error without using requireJS.

I fix this problem temporarily through adding the i18n/de.js after the select2.min.js so i can work with it till a fix/statement is available.

The only thing i use is an additional script who builds a collection.js from multiple js files and store it under the assets folder.

**Comment by JanMalte** [comment]
@MrTool an alternative temporary fix would be to set the language fix to "en". Don't see this as a final solution!
```js
$('.select2').select2({
    language: "en",
});
```

**Comment by kevin-brown** [comment]
Select2 cannot automatically load the language files for you when using RequireJS. This mostly has to do with Select2 not being able to determine the correct prefix*, which is why it falls back on `select2/i18n/` as the prefix.

If you know the path to your translation file (`de.js` in this case), you can specify it when initializing Select2.

```js
$('.select2').select2({
    language: "a/b/de",
});
```

And Select2 will try to automatically import it using that name instead of `select2/i18n/de`.

The other option, if you have a complete translation file (German is not complete), is to load in the translations directly.

```js
define(["path/to/de"], function (GermanLocale) {
  $(".select2").select2({
    language: GermanLocale
  });
});
```

This will then offload the work of dynamically loading the translation to your RequireJS loader, but at the cost of making sure it is a complete translation.

_* If anyone knows a consistent way of doing this, I'm really interested in making this work._

> I fix this problem temporarily through adding the i18n/de.js after the select2.min.js so i can work with it till a fix/statement is available.

I _think_ this is how [it is supposed to be done](https://github.com/select2/select2#internationalization-i18n) (and has been done in the past).

**Comment by MrTool** [comment]
Can you explain how select2 knows which language he should load?

If it depends on the browser language i see following "feature":
If you have a german website and work with an browser that's language is set to german, select2 shouts "hey there is no i18n/de.js", as a brave developer i add the de.js after the select2.js and no errors for me anymore. But what is if a person which browser language is france visits my site?

**Comment by kevin-brown** [comment]
https://select2.github.io/examples.html#language

It determines it from the `lang` property set within the HTML, or the `language` option when initializing Select2.

**Comment by JanMalte** [comment]
Another fix would be, if the prefix can be set as an option for select2, as setting/loading a fixed language is not really an option in my opinion.

**Comment by mateuszf** [comment]
Was this fixed ? I'm using 4.0.0-beta2
In my requirejs config "baseUrl" is set to "/static/lib" and select2 internationalization files are under "/static/lib/select2/i18n/" so relative path to lang module is: select2/i18n/*
I don't see a reason why requiring module "select2/i18n/pl" should fail but it fails !

**Comment by kevin-brown** [comment]
You need to make sure require.js isn't trying to rename the translation files and that the path that they are at is actually `select2/i18n`.
